chore: minimal values script

pameladelgado opened the pull request:
* Add a script to automatically generate a valid renku values file.
* Add script to Admin documentation.
* Reorganize Admin documentation.

ableuler reviewed (commented):
Cool, I think this will be really helpful! 
Unfortunately, mac os comes with a very outdated bash version (3.2) which can not run the script. I think you can still leave it like as it is, but describing how to run the script inside a docker container would be nice. I quickly made an image (just ubuntu + openssl). Now the command 
```
docker run -v $(pwd):/script -w /script  ableuler/ubuntu-openssl:focal ./make-values.sh -d example.com -o ./myvalues.yaml -g gitlab.com -i someid -s somesecret
```
executes the script inside a container with the current working directory mounted.

rokroskar commented:
@ableuler which mac OS? On Catalina:

```
bash --version
GNU bash, version 5.0.7(1)-release (x86_64-apple-darwin18.5.0)
```

ableuler commented:
@rokroskar I'm on Mojave. But are you sure that you're using the default bash that came with Catalina? According to a quick google search Apple stays away from newer bash versions because they don't want to use any GPLv3 projects, and are even planning to switch to `zsh` in the future.

I'd just mention that the script needs bash >=4.0 and openssl installed, and point to the provided image for those who don't want to install these dependencies locally.

rokroskar commented:
@ableuler ah no you're right - I installed it with brew 🤷

ableuler commented on docs/admin/prerequisites/dns.rst:
```diff
@@ -3,7 +3,7 @@
 ====
 
-Renku needs a DNS.
+Renku requires a DNS domain.
```
```suggestion
Renku requires a domain.
```

pameladelgado replied:
I'd leave it there, I prefer to have a bit of redundancy to avoid ambiguity.

ableuler replied:
ok

ableuler commented on docs/admin/prerequisites/certificates.rst:
```diff
@@ -27,9 +27,11 @@ For more information, please check `LetsEncrypt cert-manager Helm documentation
 
-You can also use a self-signed certificate created manually.
-Add the previously created certificate as a secret to the ``renku`` namespace.
+You can also use an SSL certificate issued by a certificate authority.
```
Maybe this could be a bit confusing as Letsencrypt is also a CA.

ableuler commented on docs/admin/index.rst:
```diff
@@ -94,105 +141,75 @@ Once all the pieces are in place, you can deploy Renku with the following comman
 ------------------
 
-If some Renku pods are not starting or present some errors please check the troubleshooting page.
+To update Renku to a newer version please check the **Upgrading from 0.x.x** section of the `release notes <https://github.com/SwissDataScienceCenter/renku/releases>`_ in case the renku-values.yaml file needs to be modified.
```
```suggestion
To update RenkuLab to a newer version please check the **Upgrading from 0.x.x** section of the `release notes <https://github.com/SwissDataScienceCenter/renku/releases>`_ in case the renku-values.yaml file needs to be modified.
```

ableuler commented on docs/admin/index.rst:
```diff
@@ -94,105 +141,75 @@ Once all the pieces are in place, you can deploy Renku with the following comman
-You can refer to the current
-`values.yaml in the renku-ui repository <https://github.com/SwissDataScienceCenter/renku-ui/blob/master/helm-chart/renku-ui/values.yaml>`_
-to check the default values and to find an example of how to apply different settings.
+#. Go to https://<renku-domain>/auth (or to your stand-alone Keycloak dashboard), login to ``Admin console`` using admin username and the decoded password stored in ``keycloak-password-secret``.
```
```suggestion
#. Go to https://<renku-domain>/auth (or to your stand-alone Keycloak dashboard), login to ``Admin console`` using "admin" as username and the decoded password stored in ``keycloak-password-secret``.
```

ableuler commented on docs/admin/index.rst:
```diff
@@ -94,105 +141,75 @@ Once all the pieces are in place, you can deploy Renku with the following comman
-2. ``layout``: The list of properties for the cookies consent plugin. We use
-   `Bootstrap <https://getbootstrap.com/docs>`_ for the UI, therefore all the Bootstrap
-   classes can be used to customize the appearance.
+If the RenkuLab deployment is using an existing GitLab, this application could be configured as an identity provider for RenkuLab's keycloak. This way all of the users from the existing GitLab instance could authenticate.
```
```suggestion
If the RenkuLab deployment is relying on an existing GitLab instance, this instance could be configured as an identity provider for RenkuLab's keycloak. This way all of the users from the existing GitLab instance can log into RenkuLab with their existing login information.
```

ableuler commented on docs/admin/index.rst:
```diff
@@ -94,105 +141,75 @@ Once all the pieces are in place, you can deploy Renku with the following comman
-for anonymous users (i.e. without an account or not currently logged in). To comply with
-international laws, it's strongly advised to explicitly require consent to the user for storing
-these data and using cookies.
+If your RenkuLab deployment includes GitLab you need to follow some additional steps to configure a user admin on GitLab.
```
```suggestion
If your RenkuLab deployment includes GitLab you need to follow some additional steps to configure an admin user on GitLab.
```

ableuler commented on docs/admin/index.rst:
```diff
@@ -94,105 +141,75 @@ Once all the pieces are in place, you can deploy Renku with the following comman
-To activate this feature, please set ``ui.privacy.enabled: true``. We have already configured a
-default cookie banner to inform the users about the aforementioned requirements and points to
-point them to the privacy page for further details.
+#. turn off automatic redirect to GitLab by setting redeploying with the value ``gitlab.oauth.autoSignIn: false``
```
```suggestion
#. turn off automatic redirection to GitLab by redeploying with the value ``gitlab.oauth.autoSignIn: false``
```

ableuler commented on docs/admin/index.rst:
```diff
@@ -94,105 +141,75 @@ Once all the pieces are in place, you can deploy Renku with the following comman
+   Troublesooting a Renku deployment <troubleshooting>
 
-.. note::
+**Note**. For deployments with existing GitLab and/or Keycloak, the post-install jobs need to be deleted manually in order for the Helm deployment to succeed.
```
Are you sure? How does this work for our CI deployments?

pameladelgado replied:
> Are you sure?

yes, we've been doing this for a couple of deployments with external GitLab and Keycloak in a regular basis. I _think_ the problem is for the keycloak deployment, which the CIs and personal deployments do have it enabled. Another issue is having the postgresql database externally, but this is a corner case and not covered in the minimal values script and template.
In any case, it seems clear that I should take out the GitLab part here.

pameladelgado replied:
> I should take out the GitLab part here

Done

rokroskar replied:
so the post-install jobs run but never finish? If that's the case it sounds like we need to fix those.

pameladelgado replied:
> so the post-install jobs run but never finish? If that's the case it sounds like we need to fix those.

They fail (iirc post-install keycloak job would never be able to connect to /auth since it has been deployed elsewhere) and as a consequence Helm thinks the deployment failed
```
Release "renku" failed: failed post-install: timed out waiting for the condition
```

So indeed we need to fix this. I created a PR that should fix this: https://github.com/SwissDataScienceCenter/renku/pull/1923 (indeed GitLab post-install job did not need this fix).

rokroskar replied:
Right, I was going to suggest a fix like that - but in that case I would remove this word of warning here since once the other PR is merged this will no longer happen. 👍

pameladelgado replied:
> Right, I was going to suggest a fix like that - but in that case I would remove this word of warning here since once the other PR is merged this will no longer happen. +1

yes, this was the intention. I'm going to delete this warning. :+1:

ableuler commented on docs/admin/index.rst:
```diff
@@ -60,31 +73,65 @@ Ensure that PVs are added correctly and have the appropriate storage class.
+   * Logged-out users can be given the permission to launch interactive sessions from public notebooks. This feature is turned off by default. For details on how to enable this feature, see the dedicated section below.
+   * A custom privacy notice and cookie banner can be configured (as of Renku 0.6.8). See the dedicated section below.
+   * Resources requests and limits can be adjusted for the diverse RenkuLab components
+   * For not independent GitLab you can configure an S3 backend for `lfs objects <https://docs.gitlab.com/ce/administration/lfs/index.html>`_ and `registry <https://docs.gitlab.com/ee/administration/packages/container_registry.html>`_, this configuration can be added in the ``gitlab`` section
```
```suggestion
   * For deployments that include GitLab you can configure an S3 backend for `lfs objects <https://docs.gitlab.com/ce/administration/lfs/index.html>`_ and `registry <https://docs.gitlab.com/ee/administration/packages/container_registry.html>`_, this configuration can be added in the ``gitlab`` section
```

ableuler commented on docs/admin/index.rst:
```diff
@@ -60,31 +73,65 @@ Ensure that PVs are added correctly and have the appropriate storage class.
+   * A custom privacy notice and cookie banner can be configured (as of Renku 0.6.8). See the dedicated section below.
+   * Resources requests and limits can be adjusted for the diverse RenkuLab components
+   * For not independent GitLab you can configure an S3 backend for `lfs objects <https://docs.gitlab.com/ce/administration/lfs/index.html>`_ and `registry <https://docs.gitlab.com/ee/administration/packages/container_registry.html>`_, this configuration can be added in the ``gitlab`` section
+   * It is possible to dedicate some nodes for running user environments.
```
```suggestion
   * It is possible to dedicate some nodes for running interactive user sessions.
```

ableuler commented on docs/admin/index.rst:
```diff
@@ -94,105 +141,75 @@ Once all the pieces are in place, you can deploy Renku with the following comman
-``ui.privacy.page`` to enable the feature and set the reference ConfigMap name and key.
-Both ``ui.privacy.enabled`` and ``ui.privacy.page.enabled`` need to be ``true`` for
-enabling the privacy page.
+   Troublesooting a Renku deployment <troubleshooting>
```
```suggestion
   Troubleshooting a RenkuLab deployment <troubleshooting>
```

ableuler commented on docs/admin/index.rst:
```diff
@@ -94,105 +141,75 @@ Once all the pieces are in place, you can deploy Renku with the following comman
 
-The `Markdown syntax <https://en.wikipedia.org/wiki/Markdown>`_ is fully supported for the
-privacy page content.
+After Renku has been deployed you can make some post deployment configurations.
```
```suggestion
After RenkuLab has been deployed you can make some post deployment configurations.
```

ableuler reviewed (changes requested):
Thanks for putting in this work and sorry for not reviewing this earlier. I've added suggestions that should avoid the login problems we've seen on external deployments recently.

I've also a bunch of small comments for the documentation part.

pameladelgado commented on charts/example-configurations/minimal-values.tmpl:
```diff
@@ -0,0 +1,277 @@
+
+  gitlabURL:                 &gitlabURL                     https://[gitlab-domain]
+  gitlabPrefix:              &gitlabPrefix                  [gitlab-prefix]           ## /gitlab if part of Renku, / otherwise
+  gitlabClientId:            &gitlabClientId                [gitlab-application-clientID]
```
This is fixed in the newest version. I did not hard-code 'jupyterhub' because the script and template should also work for external GitLab deployments.

pameladelgado commented on charts/example-configurations/minimal-values.tmpl:
```diff
@@ -0,0 +1,277 @@
+## Configuration for the Gateway service
+gateway:
+  gitlabUrl: *gitlabURL
+  gitlabClientId: *gitlabClientId
```
afaik this is needed for external GitLab deployments, so I left it there. We could revise these and the renku-gateway values when we do a chart cleanup.

pameladelgado commented on charts/example-configurations/minimal-values.tmpl:
```diff
@@ -0,0 +1,277 @@
+gateway:
+  gitlabUrl: *gitlabURL
+  gitlabClientId: *gitlabClientId
+  gitlabClientSecret: *gitlabClientSecret
```
latest version includes this and it works for external GitLab deployments

pameladelgado commented on charts/example-configurations/make-values.sh:
```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
```
mm using `env` might give unexpected results, it does not work for me.
I'd prefer to leave it as is for the moment.
There's a small section for MacOS users in the admin documentation which, together with Linux, should cover most users :slightly_smiling_face:

pameladelgado commented:
> Thanks for putting in this work and sorry for not reviewing this in detail earlier. I've added suggestions that should avoid the login problems we've seen on external deployments recently.
> 
> I've also a bunch of small comments for the documentation part.

Thanks a lot for the review. Indeed, this PR has gone stale, I'm reviving it and I incorporated some fixes I had locally. 
I also included some of your suggestions, in the documentation. :+1:

pameladelgado commented:
This is ready for review. The values generated with the script were validated for two cases: 
1. a deployment against an external GitLab, and 
2. a deployment that includes GitLab as part of the RenkuLab deployment.

pameladelgado commented on docs/admin/index.rst:
```diff
@@ -94,105 +141,75 @@ Once all the pieces are in place, you can deploy Renku with the following comman
-2. log in as the root user using the password from ``gitlab.password``
-3. modify any users you want to modify (e.g. to make them admin)
-4. turn the automatic redirect back on
+  * Helm command completed successfully.
```
I added instructions for the Helm test in a separate section. 
I find these checks to be complementary to the helm tests. Especially the quick checks, which help the admin user to understand better what could be wrong when certain functionality does not work and they are much faster/direct than running the acceptance tests.

ableuler replied:
Yep - mentioning them further down is totally sufficient for me.

ableuler reviewed (approved):
LGTM 👍
